sormas-backend/src/main/java/de/symeda/sormas/backend/dashboard/DashboardService.java

```diff
@@ -0,0 +1,371 @@
+import de.symeda.sormas.utils.CaseJoins;
+
+@Stateless
+@LocalBean
+public class DashboardService {
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	@EJB
+	private CaseService caseService;
+
+	@EJB
+	private EventService eventService;
+
+	public List<DashboardCaseDto> getCases(DashboardCriteria dashboardCriteria) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<DashboardCaseDto> cq = cb.createQuery(DashboardCaseDto.class);
+		Root<Case> caze = cq.from(Case.class);
+
+		final CaseQueryContext caseQueryContext = new CaseQueryContext(cb, cq, caze);
+		final CaseJoins<Case> joins = (CaseJoins<Case>) caseQueryContext.getJoins();
+		Join<Case, Symptoms> symptoms = joins.getSymptoms();
+		Join<Case, Person> person = joins.getPerson();
+
+		Predicate filter = caseService.createUserFilter(cb, cq, caze, new CaseUserFilterCriteria().excludeCasesFromContacts(true));
+		Predicate criteriaFilter = createCaseCriteriaFilter(dashboardCriteria, caseQueryContext);
+		filter = CriteriaBuilderHelper.and(cb, filter, criteriaFilter);
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+
+		List<DashboardCaseDto> result;
+		if (filter != null) {
+			cq.where(filter);
+			cq.multiselect(
+				caze.get(Case.ID),
+				caze.get(Case.UUID),
+				caze.get(Case.REPORT_DATE),
+				symptoms.get(Symptoms.ONSET_DATE),
+				caze.get(Case.CASE_CLASSIFICATION),
+				caze.get(Case.DISEASE),
+				caze.get(Case.INVESTIGATION_STATUS),
+				person.get(Person.PRESENT_CONDITION),
+				person.get(Person.CAUSE_OF_DEATH_DISEASE));
+
+			result = em.createQuery(cq).getResultList();
+		} else {
+			result = Collections.emptyList();
+		}
+
+		return result;
+	}
+
+	public Map<CaseClassification, Integer> getCasesCountByClassification(DashboardCriteria dashboardCriteria) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Object[]> cq = cb.createQuery(Object[].class);
+		Root<Case> caze = cq.from(Case.class);
+
+		final CaseQueryContext caseQueryContext = new CaseQueryContext(cb, cq, caze);
+
+		Predicate filter = caseService.createUserFilter(cb, cq, caze, new CaseUserFilterCriteria().excludeCasesFromContacts(true));
+		Predicate criteriaFilter = createCaseCriteriaFilter(dashboardCriteria, caseQueryContext);
+		filter = CriteriaBuilderHelper.and(cb, filter, criteriaFilter);
+
+		Map<CaseClassification, Integer> result;
+		if (filter != null) {
+			cq.multiselect(caze.get(Case.CASE_CLASSIFICATION), cb.count(caze.get(Case.CASE_CLASSIFICATION)));
+			cq.where(filter);
+			cq.groupBy(caze.get(Case.CASE_CLASSIFICATION));
+
+			result = em.createQuery(cq)
+				.getResultStream()
+				.collect(Collectors.toMap(tuple -> (CaseClassification) tuple[0], tuple -> ((Number) tuple[1]).intValue()));
+		} else {
+			result = Collections.emptyMap();
+		}
+
+		return result;
+	}
+
+	public List<DashboardQuarantineDataDto> getQuarantineData(DashboardCriteria dashboardCriteria) {
```

Comment:
Please remove this method, add the quarantine from and to dates to the DashboardCaseDto and DashboardContactDto instead, and then calculate the quarantine data based on that directly in the DashboardDataProvider; the dataset used for the quarantine data needs to be identical to the dataset that is used in the rest of the dashboard, so there's no additional query needed.